programs/zstdcli.c

```diff
@@ -544,6 +544,14 @@ static int init_cLevel(void) {
     return ZSTDCLI_CLEVEL_DEFAULT;
 }
 
+#define ZSTD_CLI_STATIC_ASSERT(c) (void)sizeof(char[(c) ? 1 : -1])
+
+#define ZSTD_NB_STRATEGIES ZSTD_STRATEGY_MAX
+
+static const char* ZSTD_strategyMap[ZSTD_NB_STRATEGIES + 1] = { "", "ZSTD_fast",
```

Comment:
OK, so this will warn you if `ZSTD_NB_STRATEGIES` is reduced in the future,
but not if it grows.

Unfortunately, growing is a more likely scenario (we may add a new strategy in the future).

Suggestion to detect any future change of `ZSTD_STRATEGY_MAX` : 
continue to separate `ZSTD_NB_STRATEGIES` from `ZSTD_STRATEGY_MAX`, and ensure their values are equal.
If you use a simple runtime `assert()`, you don't need `ZSTD_STRATEGY_MAX` : a runtime call to stable `ZSTD_cParam_getBounds()` will provide the same value.

If you use a `static_assert`, then `ZSTD_STRATEGY_MAX` is needed.